examples/ffi/exporter.cpp

```diff
@@ -15,8 +15,9 @@ struct Deleter {
   void operator()(ddog_prof_Profile *object) { ddog_prof_Profile_drop(object); }
 };
 
-template <typename T> void print_error(const char *s, const T &err) {
-  printf("%s (%.*s)\n", s, static_cast<int>(err.len), err.ptr);
+void print_error(const char *s, const ddog_Error &err) {
+  auto charslice = ddog_Error_message(&err);
+  printf("%s (%.*s)\n", s, static_cast<int>(charslice.len), charslice.ptr);
 }
```

Comment:
Minor: Since the error string is always going to be printed and accessed from the ffi client side, I wonder if it'd be nice to just make it a `char *` to save the effort of every API client having to turn it into a `char *` one way or another :)

Reply:
It's a `ddog_CharSlice`, which does use `char*` for it's pointer. We can append a null byte for convenience if you want, but that doesn't guarantee that there aren't any interior null bytes. Meaning, you need the length anyway.

Reply:
Right, what I was thinking is -- for error messages _specifically_ libdatadog could ensure that it output a real bona-fide `char *` with only a null byte at the end so that clients could print it directly.